src/main/java/org/datadog/jmxfetch/reporter/StatsdReporter.java

```diff
@@ -50,11 +50,29 @@ public void sendServiceCheck(String checkName, String status, String message,
             init();
         }
 
-        ServiceCheck sc = new ServiceCheck(String.format("%s.can_connect", checkName),
-            this.statusToInt(status), message, hostname, tags);
+        //ActiveMQ > 5.8 hack
+        String dataName = StatsdReporter.formatDataName(String.format("%s.can_connect", checkName));
+        ServiceCheck sc = new ServiceCheck(dataName, this.statusToInt(status), message, hostname, tags);
+
         statsDClient.serviceCheck(sc);
     }
 
+    public static String formatDataName(String name){
+        String[] chunks = name.split("\\.");
+        
+        // Escaping the prefix name 
+        chunks[0] = chunks[0].replaceAll("[A-Z0-9:_\\-]", "");
+
+        // Putting the chunks together (there's nothing in the standard library to do that :o )
```

Comment:
Did you have a change to try [StringUtils.join](http://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/StringUtils.html#join%28java.util.Collection,%20char%29) from Apache Commons Lang ? That would be way easier